src/easylink/pipeline_schema_constants.py

```diff
@@ -0,0 +1,165 @@
+from pathlib import Path
+
+from easylink.step import CompositeStep, ImplementedStep, InputStep, ResultStep
+from easylink.utilities.validation_utils import validate_input_file_dummy
+
+ALLOWED_SCHEMA_PARAMS = {
+    "pvs_like_case_study": {
+        "input_data_schema": {
+            "step_type": InputStep,
+            "step_params": {
+                "input_validator": lambda *_: None,
+                "out_dir": Path(),
+            },
+            "in_edges": {},
+        },
+        "pvs_like_case_study": {
+            "step_type": ImplementedStep,
+            "step_params": {
+                "input_validator": lambda *_: None,
+                "out_dir": Path("results"),
+            },
+            "in_edges": {
+                "input_data_schema": {
+                    "env_var": "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                    "files": ["file1"],
+                }
+            },
+        },
+        "results_schema": {
+            "step_type": ResultStep,
+            "step_params": {
+                "input_validator": lambda *_: None,
+                "out_dir": Path(),
+            },
+            "in_edges": {
+                "pvs_like_case_study": {
+                    "env_var": "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                    "files": ["result.parquet"],
+                },
+            },
+        },
+    },
+    "development": {
+        "input_data_schema": {
+            "step_type": InputStep,
+            "step_params": {
+                "input_validator": validate_input_file_dummy,
+                "out_dir": Path(),
+            },
+            "in_edges": {},
+        },
+        "step_1": {
+            "step_type": CompositeStep,
+            "step_params": {
+                "step_1": {
+                    "step_type": ImplementedStep,
+                    "step_params": {
+                        "input_validator": validate_input_file_dummy,
+                        "out_dir": Path("intermediate"),
+                    },
+                },
+            },
+            "in_edges": {
+                "input_data_schema": {
+                    "env_var": "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                    "files": ["file1"],
```

Comment:
Specifically, it's looking for a file keyed with "file1" in the input data configuration yaml. It occurs to me that maybe I didn't actually validate that this key exists in the config validation stage

Reply:
Ah interesting, so this type of edge (the one that comes from the pipeline's input and not from a previous step's output) is special? That seems especially hard to track given that it is coming from another "step" (input_data_schema).

Reply:
yeah, if you recall, a similar discrepancy is why in the last "slots" implementation, i split "previous step input" and "input data input" into separate parameters. I was hoping to avoid defining encapsulated abstractions for the inputs under the idea that just defining edge attributes would be good enough, but maybe having explicit interfaces for the slots would be helpful here

Reply:
so that the PipelineScehema can be instantiated independently of the input data yaml--I thought we couldn't know the "real" locations of the input files until the user defines them, which is why I resolve them into the correct filepaths as part of the PipelineGraph constructon, where we have access to the config object.

Reply:
Ah, right. Makes sense.

In other words, you define a "step" that is an `InputStep`, and then edges coming from that step are the pipeline input data. The "file" in those edges is a key in the `input_data.yaml`. For all other edges, the "file" is an actual file path in the `out_dir`.

I wonder if we could make these more similar by having "output slots" as I mentioned elsewhere, so what is now called "file" in this data structure is _always_ a semantic key into a dictionary of available data.